lib/spack/spack/mirror.py

```diff
@@ -100,6 +100,65 @@ def from_dict(d, name=None):
         else:
             return Mirror(d['fetch'], d['push'], name)
 
+    @staticmethod
+    def from_args(args):
+        result = None
+
+        if ((args.directory and args.mirror_name) or
+                (args.mirror_name and args.mirror_url) or
+                (args.mirror_url and args.directory)):
```

Comment:
Are you aware of argparse method `add_mutually_exclusive_group`?  There are other commands that make use of it, it might be useful here and easier to read/understand.

Reply:
I see you've used the built argparse approach to make sure only one of those options is provided, thanks.  Have you kept this check because of how the buildcache cmd arg handling doesn't make use of the same approach?

Reply:
I kept this as an extra level of sanity checking.  I figure that if new call sites are introduced, they may not enforce the same constraints when constructing `args`, so I decided not to assume as much...

...although I should probably update the message in the `ValueError` below since we're now one-step removed from CLI arguments.